service-base/src/main/java/fi/nls/oskari/util/IOHelper.java

```diff
@@ -828,4 +859,47 @@ public static String getParams(Map<String, String> params) {
         // drop last character ('?' or '&')
         return urlBuilder.substring(0, urlBuilder.length()-1);
     }
+
+    /**
+     * Very similiar to IOHelper.getParams(Map<String, String> params)
+     * only difference is that this method encodes the keys as well as the values
+     * @param kvps map containing the key value pairs
+     * @return String
+     */
+    public static String formURLEncode(final Map<String, String> kvps) {
```

Comment:
Probably, or maybe we could add yet another function formURLEncode(Map, boolean encodeKeys) and this would call it with (kvps, true); and getParams() would call it with (kvps, false);